multivariate_normal_model.h

```diff
@@ -2,59 +2,32 @@
 
 #include "opencv2/core/mat.hpp"
 
-/// \brief A class that stores a trained multivariate normal model.
+/// \brief Represents a multivariate normal model.
 ///
-/// Given some training samples, the class will create a multivariate normal model of the data.
-/// When successfully trained, the model can compute a transformed Mahalanobis distance that
-/// indicates on a scale [0,1] how well a test pixel fits with the model.
+/// Given some training samples, the class will train a multivariate normal model of the data.
 class MultivariateNormalModel
 {
 public:
-  /// Create a new, empty model.
-  MultivariateNormalModel();
+  /// Train a new model.
+  explicit MultivariateNormalModel(const cv::Mat& samples);
 
-  /// Tell if the model has been trained yet or not.
-  bool isTrained() const;
+  /// Compute the Mahalanobis distance between the model and a image of feature pixel vectors.
+  cv::Mat computeMahalanobisDistances(const cv::Mat& image) const;
 
-  /// Create a model given some training samples
-  void performTraining(const cv::Mat& samples);
+  /// Returns the trained mean.
+  cv::Mat mean() const;
 
-  /// If the model is already trained, compute transformed Mahalanobis distance image for all pixels in image.
-  cv::Mat computeTransformedMahalanobisImage(const cv::Mat& image) const;
+  /// Returns the trained covariance.
+  cv::Mat covariance() const;
 
-  /// Replace a ratio of samples_ with new_samples and retrain the model
-  void update(const cv::Mat& new_samples, float update_ratio);
+  /// Returns the inverse of the trained covariance.
+  cv::Mat inverseCovariance() const;
 
 private:
-  bool is_trained_;
-  cv::Mat inv_covar_;
-  cv::Mat mu_;
-  cv::Mat samples_;
+  cv::Mat mean_;
+  cv::Mat covariance_;
+  cv::Mat inverse_covariance_;
 
-  cv::Mat computeMahalanobisDistances(const cv::Mat& image) const;
+  /// Train a model given some training samples
+  void performTraining(const cv::Mat& samples);
```

Comment:
Ja dette syns jeg var en fornuftig endring av klassen. Jeg lurte fælt på denne `samples_`, men visste ikke om det lå i mitt mandat å snu om på ting da jeg så gjennom. Så bra du tok det.

Reply:
Stort problem at samples var en del av klassen? Vi må jo ta vare på dem uansett for å håndtere det adaptive tilfellet... Hvis dere har en god grunn, så kanskje jeg lærer noe nyttig. Skjønner at samplene ikke er en naturlig del av modellen, men er det eneste grunn?

Reply:
:+1: